can/typechecking.py

```diff
@@ -2,5 +2,9 @@
 """
 import typing
 
-CanFilter = typing.Dict[str, typing.Union[bool, int, str]]
+import mypy_extensions
+
+CanFilter = mypy_extensions.TypedDict(
+    "CanFilter", {"can_id": int, "can_mask": int, "extended": bool}
```

Comment:
Hmm, so looking at the CI failure, it seems like in order to take advantage of `TypedDict` we would either need to wait until it's included in `typing`, or add a runtime dependency on `mypy_extensions`?

Reply:
Oh that is much nicer than the `Union` of possible values. I'm all for adding the dependency.

Reply:
I would opt for `mypy_extensions`, since it will be added to the standard library at some point an can the be removed/it is just an intermediate solution.